sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -1022,6 +1022,18 @@ public static GpuColumnVector[] extractColumns(Table table, DataType[] colType)
     }
   }
 
+  /**
+   * Convert the table into host columns and return them, outside of a ColumnarBatch.
+   * @param colType the types of the columns.
+   */
+  public static RapidsHostColumnVector[] extractHostColumns(Table table, DataType[] colType) {
+    try (ColumnarBatch batch = from(table, colType)) {
+      return Arrays.stream(extractColumns(batch))
+          .map(GpuColumnVector::copyToHost)
```

Comment:
This may leak the RapidsHostColumnVector. 
Assume there are 10 columns, and 3 of them are copied to host. When copying the 4th column, something gets wrong and an exception is thrown out, the copied 3 columns are leaked.
We need a safeMap here, like https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/sql-plugin/src/main/scala/com/nvidia/spark/rapids/implicits.scala#L203

Reply:
I was trying to come up with an algorithm catching this. Could use another pair of eyes to verify the steps https://github.com/NVIDIA/spark-rapids/blob/c70ddde9cb59514aedac6d3acd3ca767c8f25e07/docs/dev/shims.md#exposing-an-internal-class-as-a-compile-time-dependency

Reply:
Done. Thx